utilities/generate_messages_catalog.py

```diff
@@ -131,96 +134,172 @@ def _is_callable(item):
+    source directory of each plugin to be passed to the list of roots to use
+    during the AST.walk for message inspection
+
+    :return: Return is a list of strings representing the paths to the install
+        locations of each plugin in the plugin requirements file.
+    :rtype: list [str]
+    """
+    plugin_locations = []
+    plugin_list = list(pkutils.parse_requirements(PLUGINS_FILE))
+
+    for plugin_requirement in plugin_list:
+        # Checking for a pinned requirement in the file to remove the extra
+        # information to prevent issues with the __import__ statement.
+        pin_separation_index = plugin_requirement.find("=")
+        if pin_separation_index > 0:
+            plugin_name_only = plugin_requirement[:pin_separation_index]
+        else:
+            plugin_name_only = plugin_requirement
+        plugin = __import__(plugin_name_only, globals(), locals(), [], 0)
+        plugin_location = os.path.dirname(plugin.__file__)
+        del plugin
+        plugin_locations.append(plugin_location)
+    return plugin_locations
+
+
+def _build_id_messages(python_module):
     """
     Helper function to build the messages for a given python modules out of a
     python arelle sub-package.
 
-    :param python_package: Sub-package of Arelle to build messages from.
-    :type python_package: iterable
-    :param package_root: Base path of the root of the package being parsed.
-    :type package_root: str
-    :return:
-    :rtype: list
+    :param python_module: Module location to be walked and introspected for
+        messages to build.
+    :type python_module: str
+    :return: A listing, in dictionary format, of all the messages of the
+        specified module which can then be used to generate the XML list.
+    :rtype: list [dict]
     """
     id_messages = []
-    file_count = 0
-    for python_module in python_package:
-        file_count += 1
-        full_filename_path = os.path.join(package_root, python_module)
-        refFilename = (
-            full_filename_path[len(package_root) + 1:].replace("\\", "/")
-        )
-        with open(full_filename_path, encoding="utf-8") as module_file:
-            tree = ast.parse(module_file.read(), filename=python_module)
-            callables = filter(_is_callable, ast.walk(tree))
-            for item in callables:
-                try:
-                    # imported function could be by id instead of attr
-                    handler = FUNC_HANDLER.get(
-                            item.func.attr, lambda x:  ("", 0)
-                    )
-                    if isinstance(handler, tuple):
-                        level, args_offset = handler
-                    else:
-                        level, args_offset = handler(item)
-
-                    msgCodeArg = item.args[0 + args_offset]  # str or tuple
-                    if isinstance(msgCodeArg,ast.Str):
-                        msgCodes = (msgCodeArg.s,)
+    ref_filename = os.path.basename(python_module)
```

Comment:
The `ref_filename` component is just used to create one of the fields in the xml message, specifically the module field. All opens are done with full path objects. Do you think renaming the variable to something like `ref_module` would be clearer?